pkg/server/plugin/upstreamauthority/gcpca/gcpca.go

```diff
@@ -0,0 +1,284 @@
+	return &Plugin{}
+}
+
+// SetLogger will be called by the catalog system to provide the plugin with
+// a logger when it is loaded. The logger is wired up to the SPIRE core
+// logger
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	// TODO: store the logger for later use. If the plugin does not need to
+	// log, this method can be removed.
+	p.log = log
+}
+
+func getClient(ctx context.Context) (*pcaapi.CertificateAuthorityClient, error) {
+	creds, err := google.FindDefaultCredentials(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	pcaClient, err := pcaapi.NewCertificateAuthorityClient(ctx, option.WithCredentials(creds))
+
+	if err != nil {
+		return nil, err
+	}
+
+	return pcaClient, nil
+}
+
+func pemToX509(pemStr string) (*x509.Certificate, error) {
+	block, rest := pem.Decode([]byte(pemStr))
+	if len(rest) > 0 {
+		return nil, errors.New("Unexpected trailer in certificate of size " + fmt.Sprint(len(rest)))
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return cert, nil
+}
+
+func (p *Plugin) mintX509CA_(log hclog.Logger, ctx context.Context, csr []byte, preferredTtl int32) (*upstreamauthority.MintX509CAResponse, error) {
+	csrParsed, err := x509.ParseCertificateRequest(csr)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse CSR: %v", err)
+	}
+
+	validity := time.Second * time.Duration(preferredTtl)
+	parentPath := "projects/" + p.c.Project + "/locations/" + p.c.Location + "/certificateAuthorities/" + p.c.CertName
+
+	log.Info("Minting X509 using certificate at " + parentPath + "; ttl " + validity.String())
+
+	subject := privatecapb.Subject{}
+	extractFirst := func(strings []string, into *string) {
+		if len(strings) > 0 {
+			*into = strings[0]
+		}
+	}
+
+	extractFirst(csrParsed.Subject.Organization, &subject.Organization)
+	extractFirst(csrParsed.Subject.OrganizationalUnit, &subject.OrganizationalUnit)
+	extractFirst(csrParsed.Subject.Locality, &subject.Locality)
+	extractFirst(csrParsed.Subject.Province, &subject.Province)
+
+	createRequest := privatecapb.CreateCertificateRequest{
+		Parent: parentPath,
+		Certificate: &privatecapb.Certificate{
+			Lifetime: ptypes.DurationProto(validity),
+			CertificateConfig: &privatecapb.Certificate_Config{
+				Config: &privatecapb.CertificateConfig{
+					PublicKey: &privatecapb.PublicKey{
+						Type: privatecapb.PublicKey_PEM_EC_KEY,
+						Key: pem.EncodeToMemory(
+							&pem.Block{
+								Type:  "PUBLIC KEY",
+								Bytes: csrParsed.RawSubjectPublicKeyInfo,
+							},
+						),
+					},
+					SubjectConfig: &privatecapb.CertificateConfig_SubjectConfig{
+						Subject:    &subject,
+						CommonName: csrParsed.Subject.CommonName,
+					},
+					ReusableConfig: &privatecapb.ReusableConfigWrapper{
+						ConfigValues: &privatecapb.ReusableConfigWrapper_ReusableConfig{
+							ReusableConfig: fmt.Sprintf("projects/%s/locations/%s/reusableConfigs/%s", "privateca-data", p.c.Location, "subordinate-mtls-pathlen-0"),
```

Comment:
nitpick: there is a little less cognitive overhead understanding the final result here if you place fixed values directly into the format string
```suggestion
							ReusableConfig: fmt.Sprintf("projects/privateca-data/locations/%s/reusableConfigs/subordinate-mtls-pathlen-0", p.c.Location),
```